### PR TITLE
Improve transient authorization data handling

### PR DESCRIPTION
**Pull request by joshcanhelp**
### Changes

**Potential breaking changes:**

- Remove the `session_base_name` config option for the `Auth0` class. Instead, pass the `StoreInterface` implementing class with the modified name.
- Remove `state_handler` config key, `StateHandler` interface, and `SessionStateHandler` and `DummyStateHandler` classes. This makes state handling mandatory for callback routes using `Auth0::getUser()` or `Auth0::exchange()`. Guidance for IpD-initiated flows will be provided in an upcoming migration guide.
- Remove `EmptyStore` class and the ability to pass `false` on the `store` config key. Set `persist_user` to `false` in the config key to skip all persistence. 
- Default state handling is now cookies instead of sessions. This *should* provide more flexibility but may require whitelisting cookie names (`auth0__nonce` and `auth0__state` by default) on certain managed hosts. 
- Default state key changed from `auth0__webauth_state` to `auth0__state`.
- Remove unused `\Auth0\SDK\Auth0::$URL_MAP`

### Testing

- [x] This change adds test coverage
- [x] This change has been tested on PHP 7.1.29 and 7.2.18

### Checklist

- [x] All existing and new tests complete without errors.
- [x] The correct base branch is being used - `7.0.0-dev`

### CODE REVIEW

File: src/Auth0.php [hunk 1]
```diff
@@ -317,35 +304,19 @@ public function __construct(array $config)
...
-            if ($config['store'] === false) {
-                $emptyStore = new EmptyStore();
-                $this->setStore($emptyStore);
-            } else {
-                $this->setStore($config['store']);
-            }
-        } else {
-            $sessionStore = new SessionStore($session_base_name);
-            $this->setStore($sessionStore);
+        $this->store = $config['store'] ?? null;
```
Comment (joshcanhelp) [inline src/Auth0.php]: Refactor only, no expected functional changes.

File: src/Auth0.php [hunk 2]
```diff
@@ -317,35 +304,19 @@ public function __construct(array $config)
...
+        if ($this->store === false) {
+            $this->store = new EmptyStore();
+        } else if (! $this->store instanceof StoreInterface) {
+            $this->store = new SessionStore();
```
Comment (joshcanhelp) [inline src/Auth0.php]: Refactor only, no expected functional changes.

File: src/Auth0.php [hunk 3]
```diff
@@ -317,35 +304,19 @@ public function __construct(array $config)
...
         }
 
-        $this->authStore = $config['auth_store'] ?? null;
-        if (! $this->authStore instanceof StoreInterface) {
-            $this->authStore = new CookieStore();
+        $transientStore = $config['transient_store'] ?? null;
+        if (! $transientStore instanceof StoreInterface) {
+            $transientStore = new CookieStore();
```
Comment (joshcanhelp) [inline src/Auth0.php]: Name changed from #395

File: src/Auth0.php (hunk 2 above)
Comment (stevehobbsdev) [reply src/Auth0.php]: If `$config['store']` is `null`, is the intention for it to fall into the condition on line 300? I'm assuming that the check on 298 would return `false` in this case. If so, the code isn't very clear here. At first glance I thought that the check on 298 is wrong and it should be `$this->store === null`, but after looking deeper I just have more questions.

File: src/Auth0.php (hunk 3 above)
Comment (stevehobbsdev) [reply src/Auth0.php]: Just curious - why are we falling back to an `EmptyStore` above, but not for this transient store?

File: src/Auth0.php (hunk 2 above)
Comment (joshcanhelp) [reply src/Auth0.php]: Thanks for this. I'll take a step back and look at it again. It's more clear than it was before at least 😆

File: src/Auth0.php (hunk 3 above)
Comment (joshcanhelp) [reply src/Auth0.php]: `store` is for identity artifacts (user identity, tokens) and `transientStore` is for required auth transients like nonce and state. Things work fine but are not stored if `store` is empty but things break if `transientStore` is. 

That said, I'll take another look here along with the above.

**Pull request merged**